pysat/tests/test_instruments.py

```diff
@@ -391,15 +392,22 @@ def test_download_and_load(self):
                                         inst.name, inst.tag, inst.sat_id,
                                         'was not successful.')))
 
-    # Optional support
 
-    # directory_format string
+class TestInstrumentDeprectaion():
 
-    # multiple file days
+    def setup(self):
+        """Runs before every method to create a clean testing setup"""
+        warnings.filterwarnings('always', category=DeprecationWarning)
+
+    def teardown(self):
+        """Runs after every method to clean up previous testing"""
+
+    def test_deprecated_season_date_range(self):
+        """Tests that deprecation of season_date_range is working"""
 
-    # orbit information
+        fnames = ['2009-01-01.nofile']
+        with warnings.catch_warnings(record=True) as war:
+            uts, index, date = mm_test.generate_times(fnames, '100', freq='1S')
 
-        # self.directory_format = None
-        # self.file_format = None
-        # self.multi_file_day = False
-        # self.orbit_info = None
+        assert len(war) == 1
+        assert war[0].category == DeprecationWarning
```

Comment:
Need to also test warning message, to ensure the correct deprecation warning was caught.

Reply:
Also, I added a deprecation warning evaluation function in instruments.methods.testing that can be used here.